spyder/plugins/workingdirectory/container.py

```diff
@@ -77,7 +81,61 @@ def enterEvent(self, event):
         """Set current path as the tooltip of the widget on hover."""
         self.setToolTip(self.currentText())
 
+    def focusOutEvent(self, event):
+        """Handle focus out event restoring the last valid selected path."""
+        if self.add_current_text_if_valid():
+            self.selected()
+            self.hide_completer()
+        hide_status = getattr(self.lineEdit(), 'hide_status_icon', None)
+        if hide_status:
+            hide_status()
+        super().focusOutEvent(event)
+
+    # --- Own methods
+    def valid_text(self):
+        """Get valid version of current text."""
+        directory = self.currentText()
+        file = None
+        line_number = None
+
+        if directory:
+            # Search for path/to/file.py:10 where 10 is the line number
+            match = re.fullmatch(r"(?:(\d+):)?(.+)", directory[::-1])
+            if match:
+                line_number, directory = match.groups()
+                if line_number:
+                    line_number = int(line_number[::-1])
+                directory = directory[::-1]
+
+            directory = osp.abspath(directory)
+
+            # If the directory is actually a file, open containing directory
+            if os.path.isfile(directory):
+                file = os.path.basename(directory)
+                directory = os.path.dirname(directory)
```

Comment:
```suggestion
            if osp.isfile(directory):
                file = osp.basename(directory)
                directory = osp.dirname(directory)
```

Follow the convention for `os.path` in the rest of this file.